File: 4th Semester (Java)/lab-1/src/main/java/models/FullName.java
package models;

import bank.Bank;
import exceptions.BanksException;

/**
 * Contains information about name and surname.
 */
public class FullName {
    /**
     * Name of person.
     */
    private final String name;

    /**
     * Surname of person.
     */
    private final String surname;

    /**
     * FullName constructor.
     * @param name name of the person
     * @param surname surname of the person
     * @throws BanksException if surname or/and name is empty
     */
    public FullName(String name, String surname) throws BanksException {
        if (name.isEmpty()) {
            throw new BanksException("Given value name can not be null or whitespace");
        }

        if (surname.isEmpty()) {
            throw new BanksException("Given value surname can not be null or whitespace");
        }

        this.name = name;
        this.surname = surname;
    }

    public String getName() {
        return name;
    }

    public String getSurname() {
        return surname;
    }
}

File: 3rd Semester (C#)/Lab4/Banks/Models/FullName.cs
using Banks.Tools;

namespace Banks.Models;

public class FullName
{
    public FullName(string name, string surname)
    {
        if (string.IsNullOrWhiteSpace(name))
        {
            throw new BanksException($"Given value name: {name} can not be null or whitespace");
        }

        if (string.IsNullOrWhiteSpace(surname))
        {
            throw new BanksException($"Given value surname: {surname} can not be null or whitespace");
        }

        Name = name;
        Surname = surname;
    }

    public string Name { get; }
    public string Surname { get; }
}
